Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile.psm1

```diff
@@ -0,0 +1,699 @@
+
+        [Parameter()]
+        [System.String]
+        $QrCodeContent,
+
+        [Parameter()]
+        [Microsoft.Graph.IMicrosoftGraphMimeContent]
+        $QrCodeImage,
+
+        [Parameter()]
+        [System.String[]]
+        $RoleScopeTagIds,
+
+        [Parameter()]
+        [System.DateTime]
+        $TokenCreationDateTime,
+
+        [Parameter()]
+        [System.DateTime]
+        $TokenExpirationDateTime,
+
+        [Parameter()]
+        [System.String]
+        $TokenValue,
+
+        [Parameter()]
+        [System.Boolean]
+        $WifiHidden,
+
+        [Parameter()]
+        [System.Security.SecureString]
+        $WifiPassword,
+
+        [Parameter()]
+        [ValidateSet( 'none', 'wpa', 'wep' )]
+        $WifiSecurityType,
+
+        [Parameter()]
+        [System.String]
+        $WifiSsid,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $currentInstance = Get-TargetResource @PSBoundParameters
+    $PSBoundParameters.Remove('Ensure') | Out-Null
+    $PSBoundParameters.Remove('Credential') | Out-Null
+    $PSBoundParameters.Remove('ApplicationId') | Out-Null
+    $PSBoundParameters.Remove('ApplicationSecret') | Out-Null
+    $PSBoundParameters.Remove('TenantId') | Out-Null
+    $PSBoundParameters.Remove('CertificateThumbprint') | Out-Null
+    $PSBoundParameters.Remove('ManagedIdentity') | Out-Null
+    $PSBoundParameters.Remove('AccessTokens') | Out-Null
+
+    $setParameters = Remove-M365DSCAuthenticationParameter -BoundParameters $PSBoundParameters
+
+    # CREATE
+    if ($Ensure -eq 'Present' -and $currentInstance.Ensure -eq 'Absent')
+    {
+        Write-Host "Create AndroidDeviceOwnerEnrollmentProfile: $DisplayName with Enrollment Mode: $EnrollmentMode"
+
+        $CreateParameters.remove('Id') | Out-Null
+        $CreateParameters.remove('Ensure') | Out-Null
+        $CreateParameters.Remove('Verbose') | Out-Null
+
+        New-MgAndroidDeviceOwnerEnrollmentProfile @CreateParameters
```

Comment:
This currently doesn't work since we are trying to pass in invalid parameters such as CreateDateTime, ModifiedDateTime, etc.